server/utils.go

```diff
@@ -174,3 +185,196 @@ func (w *withCode) Format(s fmt.State, verb rune) {
 		}
 	}
 }
+
+func fetchListOfResources(clusterType string, environment string, loginURL string, tenantId string) (map[string][]map[string]map[string]oputil.DataAction, error) {
+	operationsMap := map[string][]map[string]map[string]oputil.DataAction{}
+
+	apiResourcesList, err := fetchApiResources()
+	if err != nil {
+		return nil, errors.Wrap(err, "Failed to fetch list of api-resources from apiserver.")
+	}
+
+	operationsList, err := fetchDataActionsList(environment, clusterType, loginURL, tenantId)
+	if err != nil {
+		return nil, errors.Wrap(err, "Failed to fetch operations from Azure.")
+	}
+
+	for _, resList := range apiResourcesList {
+		group := "v1" // core api group
+		if resList.GroupVersion != "" && resList.GroupVersion != "v1" {
+			group = strings.Split(resList.GroupVersion, "/")[0]
+		}
+
+		if len(resList.APIResources) == 0 {
+			continue
+		}
+
+		for _, apiResource := range resList.APIResources {
+			if strings.Contains(apiResource.Name, "/") {
+				continue
+			}
+
+			actionId := clusterType
+			if group != "v1" {
+				actionId = path.Join(actionId, group)
+			}
+
+			resourceName := apiResource.Name
+
+			actionId = path.Join(actionId, resourceName)
+
+			for _, operation := range operationsList {
+				if strings.Contains(operation.Name, actionId) {
+					opNameArr := strings.Split(operation.Name, "/")
+
+					if group != "v1" {
+						// extra validation to make sure groups are the same
```

Comment:
if these are not expected, should we collect the mismatched and log in the end?

Reply:
It is expected hence added the check. For example, the resource events is under "events.k8s.io" api group and "v1" which the core api group. 
if strings.Contains(operation.Name, actionId) {
This check will return true for both event resources since the check will return true on "Microsoft.Kubernetes/connectedClusters/events/.." and "Microsoft.Kubernetes/connectedClusters/events.k8s.io/.."

The extra validation check will ensure that events.k8s.io will not get added in v1 apirgroup map as well